src/main/java/agh/ics/oop/AbstractWorldMap.java

```diff
@@ -28,33 +29,34 @@ public String toString() {
     }
 
     @Override
-    public void positionChanged(Vector2d oldPosition, Vector2d newPosition){
+    public void positionChanged(Vector2d oldPosition, Vector2d newPosition) {
         Animal animal = animalMap.get(oldPosition);
         animalMap.remove(oldPosition);
         animalMap.put(newPosition, animal);
+        this.mapBoundary.positionChanged(oldPosition, newPosition);

```

Comment:
ok, ale lepiej byłoby skorzystać z tego że to obserwator i zarejstrować sie po prostu jako listener na zwierzątko, np. w momencie dodawania go na mapę.